rotkehlchen/db/utils.py

```diff
@@ -363,3 +365,67 @@ def combine_asset_balances(balances: List[SingleDBAssetBalance]) -> List[SingleD
             new_balances = _append_or_combine(new_balances, balance)
 
     return new_balances
+
+
+def db_schema_sanity_check(
+        db_name: Literal['private', 'global'],
+        cursor: 'DBCursor',
+        script_create_tables: str,
+) -> None:
+    """Ensures that database schema is not broken. Raises DBSchemaError if anything is bad."""
+    # Replacing new lines and white spaces since they may vary if by an accident code of a
+    # db upgrade was a bit different from the one that creates new tables
+    create_script_from_constant = script_create_tables.replace(' ', '').replace('\n', '').replace('\'', '"')  # noqa: E501
+
+    # Make dictionary of table_name: table_properites. Without whitespaces. For example:
+    #   "ens_mappings": "CREATETABLEIFNOTEXISTSens_mappings(addressTEXTNOTNULLPRIMARYKEY,ens_nameTEXTUNIQUE,last_updateINTEGERNOTNULL);"  # noqa: E501
+    #
+    tables_data_from_constant = {name: properties for name, properties in re.findall(
+        pattern=r'CREATETABLEIFNOTEXISTS(.+?)\((.+?)\);',
+        string=create_script_from_constant,
+    )}
+
+    # Create same mapping for tables that are currently in the database
+    cursor.execute('SELECT name, sql FROM sqlite_master WHERE type="table"')
+    tables_data_from_db = {}
+    for (name, raw_script) in cursor:
+        raw_script = raw_script.replace(' ', '').replace('\n', '').replace('\'', '"')
+        table_properties = re.findall(
+            pattern=r'CREATETABLE.*?\((.+)\)',
+            string=raw_script,
+        )[0]
+        tables_data_from_db[name] = table_properties
+
+    # Check what tables are missing from the db
+    missing_tables = tables_data_from_constant.keys() - tables_data_from_db.keys()
+    if len(missing_tables) > 0:
+        raise DBSchemaError(
+            f'Tables {missing_tables} are missing from your {db_name} database. If you have '
+            f'manually edited db please add them back, otherwise open an issue on out github.',
+        )
+    # Check what extra tables are in the db
+    extra_tables = tables_data_from_db.keys() - tables_data_from_constant.keys()
+    if len(extra_tables) > 0:
+        raise DBSchemaError(
+            f'There are extra tables {extra_tables} in your {db_name} database. If you have '
+            f'manually edited db, please remove them, otherwise open an issue on our github.',
+        )
+    # Check structure of which tables in the database differs from the expected.
+    differing_tables_properties: Dict[str, Tuple[str, str]] = {}
+    # At this point keys of two dictionaries match
+    for table_name in tables_data_from_db:
+        if tables_data_from_db[table_name] != tables_data_from_constant[table_name]:
+            differing_tables_properties[table_name] = (
+                tables_data_from_db[table_name],
+                tables_data_from_constant[table_name],
```

Comment:
at least do for the tables in the db. for the other let us know the name of the table in the `DB_CREATE_TABLES_SCRIPT` and since is a constant in the code we can look at it easily since we know the version where this error appears

Reply:
ok